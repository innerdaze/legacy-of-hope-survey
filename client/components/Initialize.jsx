import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Box from 'grommet/components/Box'
import Form from 'grommet/components/Form'
import Header from 'grommet/components/Header'
import Heading from 'grommet/components/Heading'
import Paragraph from 'grommet/components/Paragraph'
import FormField from 'grommet/components/FormField'
import TextInput from 'grommet/components/TextInput'
import Footer from 'grommet/components/Footer'
import Button from 'grommet/components/Button'

class Initialize extends Component {
  constructor(props) {
    super(props)

    this.state = {
      apiRoot: ''
    }

    this._onApiRootFieldChange = this._onApiRootFieldChange.bind(this)
    this._onFormSubmit = this._onFormSubmit.bind(this)
  }

  handleApiRootFieldChange(e) {
    this.setState({ apiRoot: e.target.value })
  }

  handleFormSubmit(e) {
    e.preventDefault()
    this.props.onApiRootFormSubmit(this.state.apiRoot)
  }

  render() {
    return (
      <Box
        full
        justify='center'
        align='center'
        pad={{ horizontal: 'medium' }}
        >
        <Form onSubmit={this.handleFormSubmit}>
          <Header>
            <Heading>
              Welcome
            </Heading>
          </Header>
          <Paragraph size='large'>
            I see it\'s your first time here.
          </Paragraph>
          <Paragraph>
            Enter the API root URL provided by Orbis to get started.
          </Paragraph>
          <FormField error={this.props.apiRootValidationError}>
            <TextInput name='apiRoot' value={this.state.apiRoot} onDOMChange={this.handleApiRootFieldChange}/>
          </FormField>
          <Footer pad={{ vertical: 'medium' }}>
            <Button label='Continue' type='submit' onClick={this.handleFormSubmit}/>
          </Footer>
        </Form>
      </Box>
    )
  }
}

Initialize.propTypes = {
  onApiRootFormSubmit: PropTypes.func.isRequired
}

Initialize.defaultProps = {
  onApiRootFormSubmit: Function.prototype
}

export default Initialize
