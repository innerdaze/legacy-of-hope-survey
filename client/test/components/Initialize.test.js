import React from 'react'
import Initialize from '../../components/Initialize'

let wrapper, onApiRootFormSubmit

describe('Testing on Initialize...', () => {
  beforeEach(() => {
    onApiRootFormSubmit = jest.fn()

    wrapper = shallow(<Initialize onApiRootFormSubmit={onApiRootFormSubmit} />)
  })

  describe('Initialize layout', () => {
    test('Expect Initialize to have Form as parent comp and Header, some Paragraph, 2 TextInput and a Footer with a Buuton in it', () => {
      expect(wrapper.find('Box').find('Form')).toHaveLength(1)
      expect(wrapper.find('Form').find('Header')).toHaveLength(1)
      expect(
        wrapper
          .find('Form')
          .find('Header')
          .find('Heading')
      ).toHaveLength(1)
      expect(wrapper.find('Form').find('Paragraph')).toHaveLength(3)
      expect(
        wrapper
          .find('Form')
          .find('FormField')
          .find('TextInput')
      ).toHaveLength(2)
      expect(wrapper.find('Form').find('Footer')).toHaveLength(1)
      expect(
        wrapper
          .find('Form')
          .find('Footer')
          .find('Button')
      ).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on prop', () => {
    test('Expect onApiRootFormSubmit is called when Form is submitted or Button Continue is clicked ', () => {
      wrapper.find('Form').simulate('submit', {
        preventDefault() {}
      })

      expect(onApiRootFormSubmit).toBeCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on state', () => {
    test('Expect storeID state to change if TextInput storeID is change', () => {
      wrapper.find('TextInput[name="storeID"]').prop('onDOMChange', {
        preventDefault() {}
      })({
        target: {
          value: 'foo'
        }
      })

      expect(wrapper.state('storeID')).toEqual('foo')
      expect(wrapper).toMatchSnapshot()
    })

    test('Expect storeID state to change if TextInput apiRoot is change', () => {
      wrapper.find('TextInput[name="apiRoot"]').prop('onDOMChange', {
        preventDefault() {}
      })({
        target: {
          value: 'bar'
        }
      })

      expect(wrapper.state('apiRoot')).toEqual('bar')
      expect(wrapper).toMatchSnapshot()
    })
  })
})
