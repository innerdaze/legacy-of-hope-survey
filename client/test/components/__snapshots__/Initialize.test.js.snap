// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on Initialize... Initialize layout Expect Initialize to have Form as parent comp and Header, some Paragraph, 2 TextInput and a Footer with a Buuton in it 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  responsive={true}
  tag="div"
>
  <Form
    compact={false}
    fill={false}
    onSubmit={[Function]}
    pad="none"
  >
    <Header
      align="center"
      direction="row"
      pad={
        Object {
          "between": "small",
          "horizontal": "none",
          "vertical": "none",
        }
      }
      responsive={false}
    >
      <Heading
        tag="h1"
      >
        Welcome
      </Heading>
    </Header>
    <Paragraph
      size="large"
    >
      I see it's your first time here.
    </Paragraph>
    <Paragraph>
      Enter the API root URL provided by Orbis to get started.
    </Paragraph>
    <FormField
      error=""
      size="medium"
      strong={false}
    >
      <TextInput
        name="apiRoot"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Paragraph>
      Set the ID of your store.
    </Paragraph>
    <FormField
      size="medium"
      strong={false}
    >
      <TextInput
        name="storeID"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium",
        }
      }
      responsive={false}
    >
      <Button
        label="Continue"
        method="push"
        onClick={[Function]}
        type="submit"
      />
    </Footer>
  </Form>
</Box>
`;

exports[`Testing on Initialize... Test on prop Expect onApiRootFormSubmit is called when Form is submitted or Button Continue is clicked  1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  responsive={true}
  tag="div"
>
  <Form
    compact={false}
    fill={false}
    onSubmit={[Function]}
    pad="none"
  >
    <Header
      align="center"
      direction="row"
      pad={
        Object {
          "between": "small",
          "horizontal": "none",
          "vertical": "none",
        }
      }
      responsive={false}
    >
      <Heading
        tag="h1"
      >
        Welcome
      </Heading>
    </Header>
    <Paragraph
      size="large"
    >
      I see it's your first time here.
    </Paragraph>
    <Paragraph>
      Enter the API root URL provided by Orbis to get started.
    </Paragraph>
    <FormField
      error=""
      size="medium"
      strong={false}
    >
      <TextInput
        name="apiRoot"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Paragraph>
      Set the ID of your store.
    </Paragraph>
    <FormField
      size="medium"
      strong={false}
    >
      <TextInput
        name="storeID"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium",
        }
      }
      responsive={false}
    >
      <Button
        label="Continue"
        method="push"
        onClick={[Function]}
        type="submit"
      />
    </Footer>
  </Form>
</Box>
`;

exports[`Testing on Initialize... Test on state Expect storeID state to change if TextInput apiRoot is change 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  responsive={true}
  tag="div"
>
  <Form
    compact={false}
    fill={false}
    onSubmit={[Function]}
    pad="none"
  >
    <Header
      align="center"
      direction="row"
      pad={
        Object {
          "between": "small",
          "horizontal": "none",
          "vertical": "none",
        }
      }
      responsive={false}
    >
      <Heading
        tag="h1"
      >
        Welcome
      </Heading>
    </Header>
    <Paragraph
      size="large"
    >
      I see it's your first time here.
    </Paragraph>
    <Paragraph>
      Enter the API root URL provided by Orbis to get started.
    </Paragraph>
    <FormField
      error=""
      size="medium"
      strong={false}
    >
      <TextInput
        name="apiRoot"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Paragraph>
      Set the ID of your store.
    </Paragraph>
    <FormField
      size="medium"
      strong={false}
    >
      <TextInput
        name="storeID"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium",
        }
      }
      responsive={false}
    >
      <Button
        label="Continue"
        method="push"
        onClick={[Function]}
        type="submit"
      />
    </Footer>
  </Form>
</Box>
`;

exports[`Testing on Initialize... Test on state Expect storeID state to change if TextInput storeID is change 1`] = `
<Box
  align="center"
  announce={false}
  direction="column"
  focusable={true}
  full={true}
  justify="center"
  pad={
    Object {
      "horizontal": "medium",
    }
  }
  responsive={true}
  tag="div"
>
  <Form
    compact={false}
    fill={false}
    onSubmit={[Function]}
    pad="none"
  >
    <Header
      align="center"
      direction="row"
      pad={
        Object {
          "between": "small",
          "horizontal": "none",
          "vertical": "none",
        }
      }
      responsive={false}
    >
      <Heading
        tag="h1"
      >
        Welcome
      </Heading>
    </Header>
    <Paragraph
      size="large"
    >
      I see it's your first time here.
    </Paragraph>
    <Paragraph>
      Enter the API root URL provided by Orbis to get started.
    </Paragraph>
    <FormField
      error=""
      size="medium"
      strong={false}
    >
      <TextInput
        name="apiRoot"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Paragraph>
      Set the ID of your store.
    </Paragraph>
    <FormField
      size="medium"
      strong={false}
    >
      <TextInput
        name="storeID"
        onDOMChange={[Function]}
        value=""
      />
    </FormField>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium",
        }
      }
      responsive={false}
    >
      <Button
        label="Continue"
        method="push"
        onClick={[Function]}
        type="submit"
      />
    </Footer>
  </Form>
</Box>
`;
